common/src/main/java/org/opensearch/sdk/DataObject.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+
+/**
+ * A structured object used for Data Exchange.
+ */
+public interface DataObject extends ToXContentObject {
```

Comment:
Can ToXContentObject be directly used? What is the benefit of using a new interface `DataObject`?

Reply:
> Lets add unit tests to the new classes introduced.

Yes I know this, just trying to not slow up development.

> Can ToXContentObject be directly used? What is the benefit of using a new interface DataObject?

Originally I thought to use the `parse()` method to produce an instance of the object but that proved too difficult with class casting, requiring tons of mucking around with reflection, anonymous classes, etc.   I finally decided to remove that and just return the parser.  So right now there's no added benefit ... but maybe we'll want to include some multitenant stuff in it?

Again, trying to rapidly iterate here on a feature branch to avoid holding other work up.